src/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -189,14 +193,18 @@ internal DateTimeOffset GetLastWriteTime(ReadOnlySpan<char> path, bool continueO
         }
 
         internal void SetLastWriteTime(string path, DateTimeOffset time)
-            => SetAccessWriteTimes(path, null, time.ToUnixTimeSeconds());
+        {
+            long secs = time.ToUnixTimeSeconds();
+            long milli = time.ToUnixTimeMilliseconds() - secs * 1000;
+            SetAccessWriteTimes(path, null, null, secs, milli * 1_000_000);
+        }
 
         private DateTimeOffset UnixTimeToDateTimeOffset(long seconds, long nanoseconds)
         {
             return DateTimeOffset.FromUnixTimeSeconds(seconds).AddTicks(nanoseconds / NanosecondsPerTick).ToLocalTime();
         }
 
-        private void SetAccessWriteTimes(string path, long? accessTime, long? writeTime)
+        private void SetAccessWriteTimes(string path, long? accessTime, long? accessTimeNano, long? writeTime, long? writeTimeNano)
```

Comment:
It would be better (clearer, more maintainable) to move the other logic in here. Pass in DateTimeOffset? and do the math where you're using it.